pkg/controller/render/render_controller.go

```diff
@@ -431,6 +443,10 @@ func (ctrl *Controller) syncMachineConfigPool(key string) error {
 		return err
 	}
 
+	if err := apihelpers.AreMCGeneratingSubControllersCompleted(ctrl.crcLister.List, ctrl.mckLister.List, selector); err != nil {
```

Comment:
looking at this and the above IsControllerConfigCompleted, I wonder if we should add a small retry to both just so we don't randomly error.

Although I don't think the errors degrade the MCO, so it's not a very important point.

Reply:
Given that the render Controller will sync again regardless, I think `retry` functionality will here be redundant. And if it is a transient error (network issues or other conditions), many things prior to this check should fail first.